test/system/src/test/java/io/pravega/test/system/MultiControllerTest.java

```diff
@@ -160,30 +157,31 @@ public void multiControllerTest() throws ExecutionException, InterruptedExceptio
 
         log.info("Test tcp:// with no controller instances running");
         AssertExtensions.assertThrows("Should throw RetriesExhaustedException",
-                controllerClient.createScope("scope" + RandomStringUtils.randomAlphanumeric(10)),
+                controllerClientDirect.get().createScope("scope" + RandomStringUtils.randomAlphanumeric(10)),
                 throwable -> throwable instanceof RetriesExhaustedException);
         if (!DOCKER_BASED) {
             log.info("Test pravega:// with no controller instances running");
             AssertExtensions.assertThrows("Should throw RetriesExhaustedException",
-                    controllerClient.createScope("scope" + RandomStringUtils.randomAlphanumeric(10)),
+                    controllerClientDiscover.get().createScope("scope" + RandomStringUtils.randomAlphanumeric(10)),
                     throwable -> throwable instanceof RetriesExhaustedException);
         }
+
         log.info("multiControllerTest execution completed");
     }
 
     private void withControllerURIDirect() throws ExecutionException, InterruptedException {
         Assert.assertTrue(createScopeWithSimpleRetry(
-                "scope" + RandomStringUtils.randomAlphanumeric(10), controllerURIDirect.get()).get());
+                "scope" + RandomStringUtils.randomAlphanumeric(10), controllerClientDirect.get()).get());
     }
 
     private void withControllerURIDiscover() throws ExecutionException, InterruptedException {
         if (!DOCKER_BASED) {
             Assert.assertTrue(createScopeWithSimpleRetry(
-                    "scope" + RandomStringUtils.randomAlphanumeric(10), controllerURIDiscover.get()).get());
+                    "scope" + RandomStringUtils.randomAlphanumeric(10), controllerClientDiscover.get()).get());
         }
     }
 
-    CompletableFuture<Boolean> createScopeWithSimpleRetry(String scopeName, URI controllerURI) {
+    CompletableFuture<Boolean> createScopeWithSimpleRetry(String scopeName, ControllerImpl controllerClient) {
```

Comment:
Passing URI (similar to the previous version : https://git.io/vpgre) and creating controller client is better since,
a. We can test how the client behaves with changed controller instance count.
b. The intent of the last part of the test with controller instance count as 0 is to verify that if all the controllers are down then the pravega:// uri scheme will not be able to fetch the controller. But in this case since we are using the client which is already created it will contact the already running controller(with service id "controller")

Reply:
changed back to creating client every time we scale the controller instances.